pkg/controllers/node/termination/terminator/terminator.go

```diff
@@ -94,46 +109,43 @@ func (t *Terminator) Drain(ctx context.Context, node *v1.Node) error {
 	return nil
 }
 
-// getPods returns a list of evictable pods for the node
-func (t *Terminator) getPods(ctx context.Context, node *v1.Node) ([]*v1.Pod, error) {
-	podList := &v1.PodList{}
-	if err := t.kubeClient.List(ctx, podList, client.MatchingFields{"spec.nodeName": node.Name}); err != nil {
-		return nil, fmt.Errorf("listing pods on node, %w", err)
-	}
-	var pods []*v1.Pod
-	for _, p := range podList.Items {
-		// Ignore if the pod is complete and doesn't need to be evicted
-		if podutil.IsTerminal(lo.ToPtr(p)) {
-			continue
-		}
-		// Ignore if kubelet is partitioned and pods are beyond graceful termination window
-		if t.isStuckTerminating(lo.ToPtr(p)) {
-			continue
-		}
-		pods = append(pods, lo.ToPtr(p))
-	}
-	return pods, nil
-}
-
 func (t *Terminator) Evict(pods []*v1.Pod) {
-	// 1. Prioritize noncritical pods https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown
-	critical := []*v1.Pod{}
-	nonCritical := []*v1.Pod{}
+	// 1. Prioritize noncritical pods, non-daemon pods https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown
+	criticalNonDaemon := []*v1.Pod{}
+	criticalDaemon := []*v1.Pod{}
+	nonCriticalNonDaemon := []*v1.Pod{}
+	nonCriticalDaemon := []*v1.Pod{}
 	for _, pod := range pods {
 		if !pod.DeletionTimestamp.IsZero() {
 			continue
 		}
 		if pod.Spec.PriorityClassName == "system-cluster-critical" || pod.Spec.PriorityClassName == "system-node-critical" {
-			critical = append(critical, pod)
+			if podutil.IsOwnedByDaemonSet(pod) {
```

Comment:
A lot of if/else conditionals here. I'm not sure if there's a way to simplify or delegate this to a function, but maybe there's a way to simplify?

Reply:
Another way we could do this is by doing 4 lo.Filter() functions, but that'd be doing 4N operations as opposed to just iterating through the array once. 

Minimal performance impact, but I think it's better this way.